improvement: allow .videojs-component.js

kevinzang opened the pull request:
##### *To:*
@cansin @udemy/team-f @udemy/team-t 

##### *What:*
improvement: allow .videojs-component.js file extension. Team T is using it for files such as https://github.com/udemy/website-django/blob/master/static/src/udemy/js/video-player/captions-menu/captions-menu.videojs-component.js.

##### *Trello:*
None

##### *What did you test:*
I applied this change manually by updating node_modules in website-django repo. I verified we can remove `/* eslint-disable filenames/match-regex */` from the .videojs-component.js files.

##### *What dashboards will you be monitoring:*
None

Commit:
improvement: allow .videojs-component.js

cansin reviewed (approved):
LGTM. I thought we told them to use `videojs-components/` folder and use regular `.js` files. But if they want to have `.videojs-component.js` I do not have any objections.

arski commented:
I don't recall that, but I'd be happy with that too.. @practual any preference? (the videojs-component does have a nice ring to it)

cansin commented:
I think it was at one of @jillesme 's PR. But yeah, `videojs-component ` does have a nice ring, so no objections on my part.

jillesme commented:
Hello it's me, the father of the `.videojs-component` extension. I thought this name made sense as it indicates that it is not just a React component but (also) a videojs component (duh). From a developers point of view I believe it makes more sense to search for "continue-button" and see a `continue-button.videojs-component.js` rather than a `continue-button.react-component.js`.

cansin commented:
Hmm, wait a second. I never suggested using `.react-component.js` though. I am confused now. Are we putting React components inside `.videojs-.component.js` files?

cansin commented:
Iirc, we had a discussion and realized we won't need multiple React components. There was a single React component that was wrapping vanilla JS videoJS components (or sth like that) previously.

cansin commented:
Yep, current `.videojs-component.js` files does *not* have React components in them, so 👍 You freaked me there for a second @jillesme . @kevinzang carry on with this PR. It still LGTM.

jillesme commented:
Oh I see, I was still thinking about my PR (the one @cansin mentioned before). This makes sense, too. So `videojs-components` will just be video-js components and not React components. Got it!

practual commented:
Thanks. Was holding off making this request until I'd built up a sufficiently large set of file such that this change was unavoidable... but I see we're already at that point! LGTM.

Commit:
Publish

 - babel-polyfill-udemy-website@3.0.0
 - babel-preset-udemy-website@3.0.0
 - eslint-config-udemy-basics@3.1.20
 - eslint-config-udemy-react-addons@3.3.1
 - eslint-config-udemy-website@4.3.0
 - eslint-plugin-udemy@4.3.0

kevinzang commented:
Note to Team F: something weird happened with the `lerna publish`. See https://udemy.slack.com/archives/C1TGV7X6F/p1528224593000046. Somehow, it didn't compare the current release with the one before it. Consequently, the changelog contains duplicate messages. For example:

babel-polyfill-udemy-website says `include all of Babel's external helpers` under 3.0.0, but that was already released under 1.0.0.

eslint-config-udemy-website says `Add a way to blacklist import specifiers.` under 4.3.0, but that was already released under 4.2.0.
